serde_arrow/src/arrow_impl/deserialization.rs

```diff
@@ -147,8 +147,15 @@ impl BufferExtract for dyn Array {
             T::F16 => convert_primitive!(Float16Type, F16, push_u16_cast),
             T::F32 => convert_primitive!(Float32Type, F32, push_u32_cast),
             T::F64 => convert_primitive!(Float64Type, F64, push_u64_cast),
+            T::Date32 => convert_primitive!(Date32Type, Date32, push_u32_cast),
             T::Date64 => convert_primitive!(Date64Type, Date64, push_u64_cast),
             T::Decimal128(_, _) => convert_primitive!(Decimal128Type, Decimal128, push_u128_cast),
+            T::Time64(unit) => {
+                if !matches!(unit, GenericTimeUnit::Nanosecond) {
```

Comment:
Why is there a check here? But not in the arrow2 impl? As far as I can see, you simply hand out the underlying i64 without any conversion. I would expect that this works regardless of the unit? If this is related to the comment in the [arrow docs](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html#variant.Time64), also microseconds are allowed.

Reply:
Thanks I addressed this in the new commits